protos/guard/Guard.proto

```diff
@@ -0,0 +1,192 @@
+// 	rpc cancelContract(Contract) returns (Result){}  //removed for phase 2
+// }
+
+//the relationship of the objects from Guard's view
+// FileStoreMeta (1:n) Contract (1:n) payout
+//contract is the signed document between the renter and host for one shard's storage
+// the contract information contains buyer, seller, escrow, guard , storage shard's information, rent duration, amount, payment method(periodically and times)
+//ContractMeta is the information which will not be changed after proposal
+//Contract technically is the contract execution record, it contains the contractMeta and state information
+message ContractMeta{ //Contract proposed by renter, will not be changed after renter proposed
+	bytes contract_id = 1; //one id which generated by renter, and will be check unique by escrow service
+	bytes renter_address = 2; //renter's wallet address
+	bytes host_address = 3; //host’s wallet address
+	bytes shard_hash = 4; //shard’s hash
+	int32 shard_index = 5; //shard’s sequence of the original file
+	int64 shard_file_size = 6; //shard’s file size
+	bytes file_hash = 7; //original file's hash, reserved point32 back to FileStoreMeta
+	Timestamp rent_start = 8; //rent start time
+	Timestamp rent_end = 9 ; //rent end time
+	bytes guard_address = 10; //renter proposed guard's address
+	bytes escrow_address = 11; //renter proposed escrow's address
+	int64 price = 12; //the price which agreed by host and rental, on the first phase, it equal to the proposed_price
+	int64 amount = 13; //the amount which agreed by host and rental, on the first phase, it should be equal file_size*duration*agreed_price
+	int64 collaeteral_amount = 14; 
+	enum Schedule{		
+    	MONTHLY = 0;
+    	QUARTERLY = 1;
+    	ANNUALLY = 2;
+	}
+    Schedule payout_schedule = 15; // schedule for payouts
+   	int32 num_payouts = 16; // numbers of payouts
+}
+
+message Contract{ //the contract executor record, changed frequently after proposal
+	Contract contract = 1;
+	enum ContractState{
+		DRAFT = 0; //the contract was draft, but the escrow service did not received the money
+		SIGNED = 1;  // the escrow service received the money from renter, the contract was start.
+		UPLOADED = 2; //the first challenge was done, which means the shard was in the host, all pending payouts can be issued now
+		LOST = 3; // the challenge failed, the following payout should be hold or even stopped
+		CANCELED = 4; //if the shard was rebuild after LOST, or after some days of NOQUESTION, it go to CANCELED status
+		CLOSED = 5; // the all work in such contract was done
+	}
+	ContractState state = 2;
+	bytes renter_signature = 3; //renter’s signature for the sub contract
+	bytes host_signature = 4; //host’s signature from the sub contract
+	Timestamp escrow_signed_time = 5; //contract signed time
+	bytes escrow_signature = 6; //escrow's signature 
```

Comment:
This struct is not used in escrow proto. Escrow may not able to sign it for you.

Reply:
It should be the signature which escrow service return at EscrowContractResult, for us know escrow had already work for it.